### PR TITLE
[PluggableDevice] add DEVICE_DEFAULT for session/transpose ops

### PR DESCRIPTION
**Pull request by quintinwang5**
Add DEVICE_DEFAULT for session/transpose ops. This PR is for PluggableDevice.

### CODE REVIEW

**Comment by jzhoulon** [comment]
@saxenasaurabh Can you help to review it? there are also several other PRs related with DEVICE_DEFAULT, thanks.
```
https://github.com/tensorflow/tensorflow/pull/50623

https://github.com/tensorflow/tensorflow/pull/50610

https://github.com/tensorflow/tensorflow/pull/50609

https://github.com/tensorflow/tensorflow/pull/50608

https://github.com/tensorflow/tensorflow/pull/50605

https://github.com/tensorflow/tensorflow/pull/50604

https://github.com/tensorflow/tensorflow/pull/50603

https://github.com/tensorflow/tensorflow/pull/50602

https://github.com/tensorflow/tensorflow/pull/50601

https://github.com/tensorflow/tensorflow/pull/50600

https://github.com/tensorflow/tensorflow/pull/50581

https://github.com/tensorflow/tensorflow/pull/50580

https://github.com/tensorflow/tensorflow/pull/50578
```

**Comment by saxenasaurabh** [comment]
Hey @jzhoulon, 

> @saxenasaurabh Can you help to review it? there are also several other PRs related with DEVICE_DEFAULT, thanks.
> 
> ```
> https://github.com/tensorflow/tensorflow/pull/50623
> 
> https://github.com/tensorflow/tensorflow/pull/50610
> 
> https://github.com/tensorflow/tensorflow/pull/50609
> 
> https://github.com/tensorflow/tensorflow/pull/50608
> 
> https://github.com/tensorflow/tensorflow/pull/50605
> 
> https://github.com/tensorflow/tensorflow/pull/50604
> 
> https://github.com/tensorflow/tensorflow/pull/50603
> 
> https://github.com/tensorflow/tensorflow/pull/50602
> 
> https://github.com/tensorflow/tensorflow/pull/50601
> 
> https://github.com/tensorflow/tensorflow/pull/50600
> 
> https://github.com/tensorflow/tensorflow/pull/50581
> 
> https://github.com/tensorflow/tensorflow/pull/50580
> 
> https://github.com/tensorflow/tensorflow/pull/50578
> ```

Hey @jzhoulon, these are on my radar. I will try to review this week or next. Thanks for the patience!

File: tensorflow/core/kernels/transpose_op.cc [hunk 1]
```diff
@@ -91,6 +91,18 @@ REGISTER_KERNEL_BUILDER(Name("InvertPermutation")
                             .HostMemory("y"),
                         InvertPermutationOp<int64>);
 
+REGISTER_KERNEL_BUILDER(Name("InvertPermutation")
+                            .Device(DEVICE_DEFAULT)
```
Comment (penpornk) [inline tensorflow/core/kernels/transpose_op.cc:95]: Can we just replace the registration for `DEVICE_GPU` with `DEVICE_DEFAULT`? The `DEVICE_DEFAULT` rule would work for `DEVICE_GPU` too. Same for other PRs.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

File: tensorflow/core/kernels/transpose_op.cc (hunk 1 above)
Comment (quintinwang5) [reply tensorflow/core/kernels/transpose_op.cc:82]: @penpornk Hi, I've changed all of my `DEVICE_DEFAULT` PRs. Please review.

**Review by penpornk — APPROVED** [review]
Thank you for the changes!

**Pull request merged**